luigi/scheduler.py

```diff
@@ -591,6 +612,7 @@ def get_work(self, worker, host=None, **kwargs):
                               for worker in self._state.get_active_workers())
 
         tasks = list(self._state.get_pending_tasks())
+        self._state.cascade_disabled_state(tasks)
```

Comment:
(1) I don't really understand why you need to do this. We won't run a disabled task anyway if any of its dependencies are disabled right?
(2) Would be much better if `cascade_disabled_state`  was an immutable method that returns all disabled tasks. Otherwise consider the problem when a top task is "undisabled" – now you need to "undisable" all the dependencies that you just marked as disabled in its transitive closure